### ISSUE TITLE
dbt templater: local packages (symlinked via dbt deps) not found in dbt project

### ISSUE THREAD

**Issue by y-abe** [sqlfluff/sqlfluff#7481]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using the dbt templater with dbt [local packages](https://docs.getdbt.com/docs/build/packages#local-packages) (configured via `local:` in `packages.yml`), sqlfluff emits a warning `File X was not found in dbt project` for every file in the local packages and falls back to raw parsing without macro resolution.

This happens because:

1. `dbt deps` creates symlinks in `dbt_packages/` pointing to the local package directories (e.g., `dbt_packages/my_pkg` → `packages/my_pkg`).
2. dbt records the symlink path (`dbt_packages/my_pkg/models/...`) as `original_file_path` in the manifest.
3. When sqlfluff receives an input file path like `packages/my_pkg/models/my_model.sql`, it resolves the real filesystem path and compares it against the manifest's `original_file_path`.
4. Since `packages/my_pkg/...` ≠ `dbt_packages/my_pkg/...`, the match fails.
5. Even when the file is specified via the `dbt_packages/` symlink path, Python resolves the symlink to the real path before comparison, so the mismatch persists.

This effectively makes the dbt templater unable to resolve any macros for local packages.

### Expected Behaviour

The dbt templater should resolve symlinks when matching input file paths against the manifest's `original_file_path`, so that files in dbt local packages are correctly identified and their compiled SQL (with macros resolved) is used for linting.

### Observed Behaviour

```
$ sqlfluff lint packages/my_pkg/models/my_model.sql
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    File /path/to/project/packages/my_pkg/models/my_model.sql was not found in dbt project
```

The file is linted but without dbt templating — macros like `ref()`, `source()`, and package macros are not resolved.

### How to reproduce

Minimal reproduction:

```bash
mkdir -p my_project/packages/my_pkg/models

cat > my_project/dbt_project.yml << 'INNEREOF'
name: my_project
version: "1.0.0"
config-version: 2
profile: my_project
model-paths: ["models"]
INNEREOF

cat > my_project/packages.yml << 'INNEREOF'
packages:
  - local: packages/my_pkg
INNEREOF

cat > my_project/packages/my_pkg/dbt_project.yml << 'INNEREOF'
name: my_pkg
version: "1.0.0"
config-version: 2
model-paths: ["models"]
INNEREOF

cat > my_project/packages/my_pkg/models/my_model.sql << 'INNEREOF'
select 1 as id
INNEREOF

cat > my_project/.sqlfluff << 'INNEREOF'
[sqlfluff]
dialect = bigquery
templater = dbt

[sqlfluff:templater:dbt]
project_dir = .
profiles_dir = .
INNEREOF

cd my_project
dbt deps     # Creates dbt_packages/my_pkg -> packages/my_pkg symlink
dbt compile  # Generates manifest with dbt_packages/my_pkg/... paths

# Both of these produce the "not found in dbt project" warning:
sqlfluff lint packages/my_pkg/models/my_model.sql
sqlfluff lint dbt_packages/my_pkg/models/my_model.sql
```

### Dialect

BigQuery (but this issue is dialect-independent — it affects the dbt templater's path resolution logic)

### Version

- sqlfluff 4.0.4
- sqlfluff-templater-dbt 4.0.4
- dbt-core 1.10.15
- Python 3.12.12

### Configuration

```ini
[sqlfluff]
dialect = bigquery
templater = dbt

[sqlfluff:templater:dbt]
project_dir = .
profiles_dir = .
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
